lib/orocos/process.rb

```diff
@@ -1210,9 +1240,27 @@ def self.wait_running(process, timeout = nil, name_service = Orocos::CORBA.name_
             end
         end
 
-        def resolve_all_tasks(cache = Hash.new, name_service: Orocos::CORBA.name_service)
+        def ior_message_valid?(process, message)
+            message = JSON.parse(message)
```

Comment:
handle exceptions close to where you expect them to happen. This should be

~~~
begin
   message = JSON.parse(message)
rescue JSON::ParseError
   return false
end
~~~